src/modelframe.jl

```diff
@@ -102,34 +104,48 @@ end
 
 const DEFAULT_CONTRASTS = DummyCoding
 
+_levels(x::Union{CategoricalArray, NullableCategoricalArray}) = levels(x)
```

Comment:
Note to self: need to change this to `unique`, cf. https://github.com/JuliaStats/DataFrames.jl/pull/1147#discussion_r96266095.